src/object/object_primitive.c

```diff
@@ -13936,22 +14255,103 @@ mr_data_cmpdisk_varnchar (void *mem1, void *mem2, TP_DOMAIN * domain, int do_coe
   assert (str_length1 == 0xFF || str_length2 == 0xFF);
 
   or_init (&buf1, str1, 0);
-  str_length1 = or_get_varchar_length (&buf1, &rc);
+  if (str_length1 == 0xFF)
+    {
+      rc = or_get_varchar_compression_lengths (&buf1, &str1_compressed_length, &str1_decompressed_length);
+      if (rc != NO_ERROR)
+	{
+	  goto cleanup;
+	}
+
+      string1 = db_private_alloc (NULL, str1_decompressed_length + 1);
+      if (string1 == NULL)
+	{
+	  /* Error report */
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, str1_decompressed_length);
+	  goto cleanup;
+	}
+
+      rc = mr_get_compressed_data_from_buffer (&buf1, string1, str1_compressed_length, str1_decompressed_length);
+      if (rc != NO_ERROR)
+	{
+	  goto cleanup;
+	}
+
+      str_length1 = str1_decompressed_length;
+      alloced_string1 = true;
+      string1[str_length1] = '\0';
+    }
+  else
+    {
+      string1 = buf1.ptr;
+    }
+
   if (rc == NO_ERROR)
     {
       or_init (&buf2, str2, 0);
-      str_length2 = or_get_varchar_length (&buf2, &rc);
+      if (str_length2 == 0xFF)
```

Comment:
@lucianlazar2211 

I'd like to propose the code structure of `mr_data_cmpdisk_string` I suggested.